### ISSUE TITLE
No error if an extension type implements an incompatible type

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#53170]
Expected error below is not reported nor by analyzer nor by CFE
```dart
// SharedOptions=--enable-experiment=inline-class

extension type V1<T>(T id) {}

extension type V(num id) implements V1<String> {}
//                                  ^^
// [analyzer] unspecified
// [cfe] unspecified

main() {
  print(V1);
  print(V);
}
```
Tested on `Dart SDK version: 3.2.0-36.0.dev (dev) (Thu Aug 3 17:04:06 2023 -0700) on "windows_x64"` and on the edge SDK (Aug 10, 2023) on Linux x64

**Comment by sgrekhov** [comment]
There is even simpler example below. Again, no error reported by analyzer and CFE
```dart
extension type V1(String id) {}

extension type V(num id) implements V1 {}
//             ^
// [analyzer] unspecified
// [cfe] unspecified

main() {
  print(V1);
  print(V);
}
```

**Comment by scheglov** [comment]
Fixed in 53cccf9c8bd76930d49c0a2e14b6b173f2dc5181, in the analyzer.
